### ISSUE TITLE
Some unknown garbage.

### ISSUE THREAD

**Issue by AlexanderCurl** [callofduty4x/CoD4x_Server#162]
Okay since @IceNinjaman said if I spot this again, need to investigate this issue.
So, yesterday I spotted a player without name and -37243284632874632 score (1st. pic).
I checked on console there was a zombi client connected on the server.

The last 2 pics is a guy who using some cyrillic chars in his name I guess.

Pics: https://imgur.com/a/jmdeW

**Comment by AlexanderCurl** [comment]
Oh I forgot, @IceNinjaman said, he checked with HLSW as well, on there looks good.

**Comment by T-Maxxx** [comment]
There was no errors like this in vanilla game so it must be something on server side.

**Comment by D4edalus** [comment]
guess we will keep that in the back of our heads. doesn't seem critical to me. let us know when anyones able to reproduce errors like that. the title is extremely undescriptive, i've changed it.

**Comment by cyberplant** [comment]
I bet it's because UTF-8 characters use more than one byte in ASCII representation, so some things relying on strings with fixed size are getting unexpected input and this unintended behaviour..

**Comment by D4edalus** [comment]
yeah i had that thought too. its probably a playername coming from steam as utf8. dont know the encoding of playernames, but i guess its just ascii. 
it could be that writing the name coming from steam to the respective string on the server causes a little bufferoverflow. (utf8 stringlength vs ascii stringlength)

**Comment by AlexanderCurl** [comment]
UTF-8 can be supported in the future?

**Comment by T-Maxxx** [comment]
I don't think so. There are a lot of places where we will have to use 2 instead of 1 for string length.

**Comment by AlexanderCurl** [comment]
Ok I leave it open, because sometimes I still spot a weird playername and activity.

**Comment by D4edalus** [comment]
UTF-8 should be supported in what way? the server *should* not have a problem with it because strings are just zero terminated. it might render some garbage on the console, but who cares really. same on the client. But the topic of this issue is rather to fix the caused errornous behaviour you described in your initial post, we don't even know if that's caused by utf8 texts, that's just speculation. 
However, if you can create a reproducable error scenario we can start fixing it.

**Comment by kreator-eb** [comment]
Still happends from time to time, if you reconnect it will fix it, if you check clients with rcon they are all mostly good but client names get messed up sometimes, people take different peoples name or they dont have a name at all, and its kinda client sided because a reconnect will fix this mostly, doesnt happend to everyone and its not common really. After disabling steam names I dont think I've noticed this issue so far.

**Comment by AlexanderCurl** [comment]
Closing as wontfix.
